be/src/io/tools/file_cache_microbench.cpp

```diff
@@ -0,0 +1,1222 @@
+
+    // Job执行结果统计
+    struct Statistics {
+        double peak_write_iops;
+        double peak_read_iops;
+        int64_t cache_hits;
+        int64_t cache_misses;
+        int64_t bytes_read_local;
+        int64_t bytes_read_remote;
+        std::string total_write_time;
+        std::string total_read_time;
+    } stats;
+
+    // 记录与作业相关的文件信息
+    std::vector<FileInfo> file_records; // 记录文件信息
+
+    // 添加 completion_tracker
+    std::shared_ptr<FileCompletionTracker> completion_tracker;
+
+    // 默认构造函数
+    Job() : job_id(""), config(), status(JobStatus::PENDING),
+            create_time(std::chrono::system_clock::now()),
+            completion_tracker(std::make_shared<FileCompletionTracker>()) {}
+
+    // 带参数的构造函数
+    Job(const std::string& id, const JobConfig& cfg)
+        : job_id(id), config(cfg), status(JobStatus::PENDING),
+          create_time(std::chrono::system_clock::now()) {
+            if (config.write_iops && config.read_iops) {
+                completion_tracker = std::make_shared<FileCompletionTracker>();
+            } else {
+                completion_tracker = nullptr;
+            }
+          }
+};
+
+// Job管理器
+class JobManager {
+public:
+    JobManager() : _next_job_id(0), _job_executor_pool(4) {} // 创建4个线程的执行池
+
+    std::string submit_job(const JobConfig& config) {
+        try {
+            std::lock_guard<std::mutex> lock(_mutex);
+            std::string job_id = "job_" + std::to_string(std::time(nullptr)) + "_" +
+                                 std::to_string(_next_job_id++);
+            _jobs.emplace(job_id, Job(job_id, config));
+
+            // 将作业提交到执行线程池
+            _job_executor_pool.enqueue([this, job_id]() {
+                try {
+                    {
+                        std::lock_guard<std::mutex> lock(_mutex);
+                        _jobs[job_id].status = JobStatus::RUNNING;
+                        _jobs[job_id].start_time = std::chrono::system_clock::now();
+                    }
+
+                    execute_job(job_id);
+
+                    {
+                        std::lock_guard<std::mutex> lock(_mutex);
+                        _jobs[job_id].status = JobStatus::COMPLETED;
+                        _jobs[job_id].end_time = std::chrono::system_clock::now();
+                    }
+                } catch (const std::exception& e) {
+                    std::lock_guard<std::mutex> lock(_mutex);
+                    _jobs[job_id].status = JobStatus::FAILED;
+                    _jobs[job_id].error_message = e.what();
+                    _jobs[job_id].end_time = std::chrono::system_clock::now();
+                    LOG(ERROR) << "Job " << job_id << " failed: " << e.what();
+                }
+            });
+
+            return job_id;
+        } catch (const std::exception& e) {
+            LOG(ERROR) << "Error submitting job: " << e.what();
+            // 返回错误信息
+            return "{\"error\": \"" + std::string(e.what()) + "\"}";
```

Comment:
use R"()" to eleminate escaping https://en.cppreference.com/w/cpp/language/string_literal